repos/spack_repo/builtin/packages/kokkos/package.py

```diff
@@ -288,14 +288,21 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("+wrapper", when="~cuda")
     conflicts("+wrapper", when="+cmake_lang")
 
-    cxxstds = ["11", "14", "17", "20"]
-    variant("cxxstd", default="17", values=cxxstds, multi=False, description="C++ standard")
+    cxxstds = ["14", "17", "20", "23"]
+    variant(
+        "cxxstd",
+        default="17",
```

Comment:
Discussion WG: we probably want 20 as default
```suggestion
        default="20",
```

Reply:
I feel a little bit sceptical about this. We were very conservative trying not to change default for older releases and here, you want to push legacy version to default on c++ 20.
Kokkos 5 will end up using 20 as default thanks to conflict but I am not sure that we want Kokkos < 5 to be c++ 20 by default.

Reply:
I need to test, I wonder what happens when the default value is not part of the possible values...

Reply:
> I feel a little bit sceptical about this. We were very conservative trying not to change default for older releases and here, you want to push legacy version to default on c++ 20. Kokkos 5 will end up using 20 as default thanks to conflict but I am not sure that we want Kokkos < 5 to be c++ 20 by default.

Isn't the std something that ultimately comes from the user or other packages. If we are able to build with >=17 for version 4, any other setting in someone's build environment that wants 17 would be fine right? For standalone builds with just Kokkos we would use 20, but given that we want people to transition to 5 eventually, nudging them in this direction looks like a good thing to me

Reply:
```
spack spec --fresh kokkos@5
 -   kokkos@5.0.0~aggressive_vectorization~cmake_lang+complex_align~cuda~debug~debug_bounds_check~hip_relocatable_device_code~hpx~hwloc~ipo~openmp~openmptarget~pic~rocm+serial+shared~sycl~tests~threads~tuning~wrapper build_system=cmake build_type=Release cxxstd=23 generator=make intel_gpu_arch=none platform=linux os=ubuntu20.04 target=icelake %cxx=gcc@10.5.0
```

`cxxstd=23` wut ? @cedricchevalier19 could you try ?

Reply:
I agree with @cedricchevalier19 that it might be too aggressive to change the default for older versions. Kokkos 5 already defaults to 20, that's good enough for future development.

Reply:
> ```
> spack spec --fresh kokkos@5
>  -   kokkos@5.0.0~aggressive_vectorization~cmake_lang+complex_align~cuda~debug~debug_bounds_check~hip_relocatable_device_code~hpx~hwloc~ipo~openmp~openmptarget~pic~rocm+serial+shared~sycl~tests~threads~tuning~wrapper build_system=cmake build_type=Release cxxstd=23 generator=make intel_gpu_arch=none platform=linux os=ubuntu20.04 target=icelake %cxx=gcc@10.5.0
> ```
> 
> `cxxstd=23` wut ? @cedricchevalier19 could you try ?

With the default value to 20, this does not happen.

Reply:
otherwise:
```
variant("cxxstd", default="17", values=("14", "17", "20",), when="@3", multi=False, description="C++ standard",)
variant("cxxstd", default="17", values=("17", "20", "23"), when="@4", multi=False, description="C++ standard",)
variant("cxxstd", default="20", values=("20", "23"), when="@5", multi=False, description="C++ standard",)
```

Reply:
@tpadioleau while explicit, it might be the best solution.

Reply:
I have updated the package, let me know what you think